cpp/src/parquet/types.h

```diff
@@ -670,6 +671,19 @@ inline std::string format_fwf(int width) {
   return ss.str();
 }
 
+template <size_t Width>
+struct UnsignedTypeWithWidth;
+
+template <>
+struct UnsignedTypeWithWidth<4> {
+  using type = uint32_t;
+};
+
+template <>
+struct UnsignedTypeWithWidth<8> {
+  using type = uint64_t;
+};
```

Comment:
Does this need to be public?

Reply:
No, will move it to encoding.cc in an unnamed namespace.